src/java.base/share/classes/java/nio/Direct-X-Buffer.java.template

```diff
@@ -75,15 +77,31 @@ class Direct$Type$Buffer$RW$$BO$
 
 #if[byte]
 
-    private record Deallocator(long address, long size, int capacity) implements Runnable {
-        private Deallocator {
+    private static final class Deallocator implements Runnable {
+
+        private static final long INVOKED_OFFSET = Unsafe.getUnsafe()
```

Comment:
There are two invocations of Unsafe.getUnsafe  introduced here, while the base class Buffer has initialised the static UNSAFE (i.e. the reference used for freeMemory.)   Could this static be used to replace Unsafe.getUnsafe ?

Reply:
Good catch!